src/Schema/Keywords/Type.php

```diff
@@ -33,70 +32,68 @@ class Type extends BaseKeyword
-                    }
-                    break;
-                case CebeType::OBJECT:
-                    if (! is_object($data) && ! is_array($data)) {
-                        throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::OBJECT, $data);
-                    }
-                    break;
-                case 'array':
-                    // no constant here yet https://github.com/cebe/php-openapi/pull/24
-                    if (! is_array($data) || ArrayHelper::isAssoc($data)) {
-                        throw TypeMismatch::becauseTypeDoesNotMatch('array', $data);
-                    }
-                    break;
-                case CebeType::NUMBER:
-                    if (! is_numeric($data)) {
-                        throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::NUMBER, $data);
-                    }
-                    break;
-                case CebeType::INTEGER:
-                    if (! is_int($data)) {
-                        throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::INTEGER, $data);
-                    }
-                    break;
-                case CebeType::STRING:
-                    if (! is_string($data)) {
-                        throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::STRING, $data);
-                    }
-                    break;
-                default:
-                    throw TypeMismatch::becauseTypeIsNotKnown($type);
-            }
+        switch ($type) {
+            case CebeType::BOOLEAN:
+                if (! is_bool($data)) {
+                    throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::BOOLEAN, $data);
+                }
+                break;
+            case CebeType::OBJECT:
+                if (! is_object($data) && ! is_array($data)) {
+                    throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::OBJECT, $data);
+                }
+                break;
+            case 'array':
+                // no constant here yet https://github.com/cebe/php-openapi/pull/24
+                if (! is_array($data) || ArrayHelper::isAssoc($data)) {
+                    throw TypeMismatch::becauseTypeDoesNotMatch('array', $data);
+                }
+                break;
+            case CebeType::NUMBER:
+                if (! is_numeric($data)) {
+                    throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::NUMBER, $data);
+                }
+                break;
+            case CebeType::INTEGER:
+                if (! is_int($data)) {
+                    throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::INTEGER, $data);
+                }
+                break;
+            case CebeType::STRING:
+                if (! is_string($data)) {
+                    throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::STRING, $data);
+                }
+                break;
+            default:
+                throw InvalidSchema::becauseTypeIsNotKnown($type);
```

Comment:
Now it's also invalid schema situation, so changed the exception

Reply:
Makes sense